test/e2e/framework/pods.go

```diff
@@ -29,70 +29,74 @@ import (
 	. "github.com/onsi/gomega"
 )
 
-// TODO: Consolidate pod-specific framework functions here.
-
 // Convenience method for getting a pod client interface in the framework's namespace.
-func (f *Framework) PodClient() unversioned.PodInterface {
-	return f.Client.Pods(f.Namespace.Name)
+func (f *Framework) PodClient() *PodClient {
+	return &PodClient{
+		f:            f,
+		PodInterface: f.Client.Pods(f.Namespace.Name),
+	}
 }
 
-// Create a new pod according to the framework specifications, and wait for it to start.
-// Returns the server's representation of the pod.
-func (f *Framework) CreatePod(pod *api.Pod) *api.Pod {
-	p := f.CreatePodAsync(pod)
-	ExpectNoError(f.WaitForPodRunning(p.Name))
-	return p
+type PodClient struct {
+	f *Framework
+	unversioned.PodInterface
 }
 
-// Create a new pod according to the framework specifications (don't wait for it to start).
-// Returns the server's representation of the pod.
-func (f *Framework) CreatePodAsync(pod *api.Pod) *api.Pod {
-	f.MungePodSpec(pod)
-	p, err := f.PodClient().Create(pod)
+// Create creates a new pod according to the framework specifications (don't wait for it to start).
+func (c *PodClient) Create(pod *api.Pod) *api.Pod {
+	c.MungeSpec(pod)
+	p, err := c.PodInterface.Create(pod)
 	ExpectNoError(err, "Error creating Pod")
 	return p
 }
 
-// Batch version of CreatePod. All pods are created before waiting.
-// Returns a slice, in the same order as pods, containing the server's representations of the pods.
-func (f *Framework) CreatePods(pods []*api.Pod) []*api.Pod {
+// CreateSync creates a new pod according to the framework specifications, and wait for it to start.
+func (c *PodClient) CreateSync(pod *api.Pod) *api.Pod {
```

Comment:
The reason I called this CreatePods, and the other CreatePodsAsync is that I think this should be the default in most cases. I'm not sure what the best naming is, but I do want to encourage this method to be used over the bare Create above.